src/iomap.cpp

```diff
@@ -384,81 +261,148 @@ bool IOMap::parseTileArea(OTB::Loader& loader, const OTB::Node& tileAreaNode, Ma
 
 		map.setTile(x, y, z, tile);
 	}
-	return true;
 }
 
-bool IOMap::parseTowns(OTB::Loader& loader, const OTB::Node& townsNode, Map& map)
+void parseTowns(const OTB::Node& townsNode, Map& map)
 {
-	for (auto& townNode : townsNode.children) {
-		PropStream propStream;
-		if (townNode.type != OTBM_TOWN) {
-			setLastErrorString("Unknown town node.");
-			return false;
+	for (auto& node : townsNode.children) {
+		if (node.type != OTBM_TOWN) [[unlikely]] {
+			throw std::invalid_argument(fmt::format("Unknown town node: {:d}.", static_cast<uint16_t>(node.type)));
 		}
 
-		if (!loader.getProps(townNode, propStream)) {
-			setLastErrorString("Could not read town data.");
-			return false;
-		}
-
-		uint32_t townId;
-		if (!propStream.read<uint32_t>(townId)) {
-			setLastErrorString("Could not read town id.");
-			return false;
-		}
+		auto first = node.propsBegin, last = node.propsEnd;
+		auto townId = OTB::read<uint32_t>(first, last);
 
 		Town* town = map.towns.getTown(townId);
 		if (!town) {
 			town = new Town(townId);
 			map.towns.addTown(townId, town);
 		}
 
-		auto [townName, ok] = propStream.readString();
-		if (!ok) {
-			setLastErrorString("Could not read town name.");
-			return false;
-		}
-
+		auto townName = OTB::readString(first, last);
 		town->setName(townName);
 
-		OTBM_Destination_coords town_coords;
-		if (!propStream.read(town_coords)) {
-			setLastErrorString("Could not read town coordinates.");
-			return false;
-		}
-
-		town->setTemplePos(Position(town_coords.x, town_coords.y, town_coords.z));
+		auto [x, y, z] = read_coords(first, last);
+		town->setTemplePos({x, y, z});
 	}
-	return true;
 }
 
-bool IOMap::parseWaypoints(OTB::Loader& loader, const OTB::Node& waypointsNode, Map& map)
+void parseWaypoints(const OTB::Node& waypointsNode, Map& map)
 {
 	PropStream propStream;
```

Comment:
unused?

Reply:
Unused. I wonder how gcc didn't catch it